modules/extensions/src/collide/collide-pass.ts

```diff
@@ -0,0 +1,45 @@
+import {Framebuffer, withParameters} from '@luma.gl/core';
+import {_LayersPass as LayersPass, LayersPassRenderOptions} from '@deck.gl/core';
+
+type CollidePassRenderOptions = LayersPassRenderOptions & {};
+
+export default class CollidePass extends LayersPass {
+  renderCollideMap(target: Framebuffer, options: CollidePassRenderOptions) {
+    const gl = this.gl;
+
+    const colorMask = [true, true, true, true];
+    const padding = 1;
+
+    return withParameters(
+      gl,
+      {
+        scissorTest: true,
+        scissor: [padding, padding, target.width - 2 * padding, target.height - 2 * padding],
+        clearColor: [0, 0, 0, 0],
+        blend: false,
+        colorMask,
+        depthTest: true,
+        depthRange: [0, 1]
+      },
+      () => this.render({...options, target, pass: 'collide'})
+    );
+  }
+
+  shouldDrawLayer(layer) {
+    return (
```

Comment:
I still don't get why you need this check. In what scenario will this check fail?

- CollideEffect already filters the layers list by checking `collideEnabled` so no disabled  layers could be passed into this step. 
- We don't look for a particular extension anywhere else, the existence of this prop is good enough indication.

Reply:
Removed `shouldDrawLayer` - I think you're right